docs/tasks/administer-cluster/kubeadm-upgrade-1-8.md

```diff
@@ -0,0 +1,254 @@
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-apiserver
+[upgrade/staticpods] Component "kube-apiserver" upgraded successfully!
+[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-controller-manager.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests155856668/kube-controller-manager.yaml"
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-controller-manager
+[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
+[upgrade/staticpods] Moved upgraded manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests155856668/kube-scheduler.yaml"
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-scheduler
+[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
+[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
+[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
+[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
+[addons] Applied essential addon: kube-dns
+[addons] Applied essential addon: kube-proxy
+
+[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.8.0". Enjoy!
+
+[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
+```
+
+`kubeadm upgrade apply` does the following:
+
+- It checks that your cluster is in an upgradeable state, that is:
+  - The API Server is reachable,
+  - All nodes are in the `Ready` state, and
+  - The control plane is healthy
+- It enforces the version skew policies.
+- It makes sure the control plane images are available or available to pull to the machine.
+- It upgrades the control plane components or rollbacks if any of them fails to come up.
+- It applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
+
+5. Manually upgrade your Software Defined Network (SDN).
+
+   Your Container Network Interface (CNI) provider might have its own upgrade instructions to follow now.
+   Check the [addons](/docs/concepts/cluster-administration/addons/) page to
+   find your CNI provider and see if there are additional upgrade steps
+   necessary.
+
+## Upgrading your worker nodes
+
+For each worker node (referred to as `$WORKER` below) in your cluster, upgrade `kubelet` by executing the following commands:
+
+1. Prepare the node for maintenance, marking it unschedulable and evicting the workload:
+
+```shell
+$ kubectl cordon $WORKER
+$ kubectl drain $WORKER
```

Comment:
I think `drain` is cordoning the node.

Reply:
Yeah @kargakis that was my hunch as well but since @luxas and I were not 100% sure about it we decided to go with the 'safe' option. Happy to change, though …